en/docs/extended_topics/python_kernel_op.md

```diff
@@ -0,0 +1,291 @@
+            *ctx->Shape4ArgNameAndIndex("in", 0);
+        *ctx->Dtype4ArgNameAndIndex("out", 0) =
+            *ctx->Dtype4ArgNameAndIndex("in", 0);
+        return Maybe<void>::Ok();
+      });
+}  // namespace
+}  // namespace oneflow
+```
+
+Analyze the above code: 
+
+- The `oneflow/core/framework/framework.h` contains all the interfaces we need to create an Op
+- `.Attr<std::string>("device_sub_tag", "py")` is necessary which tells OneFlow to call the Python Kernel when using this Op.
+- The interface related to the custom op is organized in `oneflow::user_op`. Using the namespace `oneflow` to simplify the type name.
+- The macro `REGISTER_USER_OP` is used to register Op which accepts an parameter `user_relu_forward` as `op_type_name`.
+- After registering with `REGISTER_USER_OP`, it actually returns an `OpRegistry` class (located in [user_op_registry.h]()). By calling this class, the setting of a custom op is accomplished:
+    1. `Input("in") ` indicates it has an input named "in".
+    2. `Output("out") ` indicates that it has an output named "out".
+    3. `SetTensorDescInferFn` is used to set the shape and data type derivation function. Also to describe the relationship between the shape and type of the output of the operator and the input. In the above code, the output shape and data type are same as the input.
+
+The `op_type_name_cpp_def.cpp` is the only C++ file that will be used in the implementation of the `Python Kernel`. It is used to configure information of Op. At this stage, it is not possible to skip the step of configuring Op using C++ (since it is necessary for setting advanced information such as distributions). But as we can see, the file does not involve specific operations. Even if you are not familiar with C++, you can easily master it according to our examples.
+
+## Wrapping Python Interface of Op
+In order for the user to use the `user_relu` we just created at the Python level. We need to create a `user_relu_py_api.py` file with the following contents:
+
+```python
+import oneflow as flow
+
+def user_relu_forward(x):
+    op = (
+        flow.user_op_builder("myrelu")
+        .Op("user_relu_forward")
+        .Input("in", [x])
+        .Output("out")
+        .Build()
+    )
+    return op.InferAndTryRun().SoleOutputBlob()
+```
+
+`flow.user_op_builder("op_myrelu")` actually returns a `UserOpConfBuilder` object named `op_myrelu`.
+
+This object contains methods such as `Op`, `Input` and etc which are used to encapsulate a custom op. Explained below:
+
+- `Op("user_relu_forward")`：The parameter must be `op_type_name` which previously registered in C++. OneFlow finds the Op type already registered and instantiates the op object.
+- `Input("in", [input_blob])`：Corresponds to `Input` for op registration in C++. The first parameter string must match the string set in `Input` when the op is registered in C++. The second parameter is the tensor of the input which is a `list`. Because an Op allows multiple inputs.
```

Comment:
```suggestion
- `Input("in", [input_blob])`：Corresponds to `Input` for op registration in C++. The first parameter string must match the string set in `Input` when the op is registered in C++. The second parameter is the tensor of the input which is a `list` because an Op allows multiple inputs.
```